### PR TITLE
Ovms instance fix

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

File: src/config.hpp [hunk 1]
```diff
@@ -285,6 +285,15 @@ class Config {
         return result->operator[]("metrics_enabled").as<bool>();
     }
 
+    /**
+     * @brief Check if result was parsed
+     *
+     * @return bool
+     */
+    bool isResultParsed() {
```
Comment (dkalinowski) [inline src/config.hpp:293]:
```suggestion
    bool isParsed() {
```
Result cannot be parsed, result is available once parsing is complete.

File: src/config.hpp [hunk 2]
```diff
@@ -285,6 +285,15 @@ class Config {
...
+        return result.get() == nullptr;
```
Comment (dkalinowski) [inline src/config.hpp:294]:
```suggestion
        return result;
```
Should be sufficient

File: src/metric_config.cpp [hunk 3]
```diff
@@ -36,7 +36,7 @@ bool MetricConfig::validateEndpointPath(const std::string& endpoint) {
 }
 
 // Getting the "monitoring" metrics config as input
-Status MetricConfig::parseMetricsConfig(const rapidjson::Value& metrics, uint64_t restPort) {
+Status MetricConfig::parseMetricsConfig(const rapidjson::Value& metrics, bool failIfMetricsAreEnabled) {
```
Comment (dkalinowski) [inline src/metric_config.cpp:39]:
```suggestion
Status MetricConfig::parseMetricsConfig(const rapidjson::Value& metrics, bool forceFailureIfMetricEnabled) {
```
Also add comment why it is done this way

File: src/metric_config.cpp [hunk 4]
```diff
@@ -112,7 +112,7 @@ void MetricConfig::setDefaultMetricsTo(bool enabled) {
     }
 }
 
-Status MetricConfig::loadFromCLIString(bool isEnabled, const std::string& metricsList, uint64_t restPort) {
+Status MetricConfig::loadFromCLIString(bool isEnabled, const std::string& metricsList, bool failIfMetricsAreEnabled) {
```
Comment (dkalinowski) [inline src/metric_config.cpp:115]: Why do we need `failIfMetricsAreEnabled` here? We can pass `false` in line 143, right?

Comment (dkalinowski) [reply src/metric_config.cpp:115]: If so, use default parameter

Comment (rasapala) [reply src/metric_config.cpp:115]: Done.

File: src/metric_config.cpp (hunk 3 above)
Comment (rasapala) [reply src/metric_config.cpp:39]: Done.

File: src/config.hpp (hunk 2 above)
Comment (rasapala) [reply src/config.hpp:294]: Done.

File: src/config.hpp (hunk 1 above)
Comment (rasapala) [reply src/config.hpp:293]: Done.

**Review by dtrawins — APPROVED** [review]

**Review by dkalinowski — APPROVED** [review]

File: src/config.cpp [hunk 5]
```diff
@@ -155,7 +155,7 @@ Config& Config::parse(int argc, char** argv) {
                 cxxopts::value<bool>()->default_value("false"),
                 "METRICS")
             ("metrics_list",
-                "List of metrics families to be enabled. If empty the default metrics are enabled: ovms_requests_success, ovms_requests_fail, ovms_request_time_us, ovms_streams, ovms_inference_time_us, ovms_wait_for_infer_req_time_us. Additional families: ovms_infer_req_queue_size, ovms_infer_req_active.",
+                "Comma separated list of metrics. If unset, there is assumed a default list. When set, only the listed metrics will be enabled. Default metrics: ovms_requests_success, ovms_requests_fail, ovms_request_time_us, ovms_streams, ovms_inference_time_us, ovms_wait_for_infer_req_time_us; Additional metrics: ovms_infer_req_queue_size, ovms_infer_req_active.",
```
Comment (atobiszei) [inline src/config.cpp:158]: Don't we want to have by the default metrics used in HPA in K8S? I assume for those we require :
ovms_current_requests", "ovms_infer_req_active"?
@dkalinowski / @dtrawins

Comment (rasapala) [reply src/config.cpp:158]: We need  ovms_current_requests + ovms_streams for Autoscalling.

**Pull request merged**